File: src/main/java/no/kristiania/pgr301/eksamen/entity/VehicleEntity.java
package no.kristiania.pgr301.eksamen.entity;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Entity
public class VehicleEntity {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank @Size(max = 70)
    private String name;

    @NotBlank @Size(max = 70)
    private String model;

    public VehicleEntity() {}

    public VehicleEntity(Long id, String name, String model) {
        this.id = id;
        this.name = name;
        this.model = model;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }
}

File: src_bak/main/kotlin/no/kristiania/pgr301/eksamen/entity/VehicleEntity.kt
package no.kristiania.pgr301.eksamen.entity

import java.time.ZonedDateTime
import javax.persistence.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Table(name = "vehicles")
@Entity
class VehicleEntity (
        @get:Id @get:GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @get:NotBlank @get:Size(max = 70)
        var name: String,

        @get:NotBlank @get:Size(max = 70)
        var model: String,

        @get:NotNull
        var creationTime: ZonedDateTime? = null,

        @get:NotNull
        var updated: ZonedDateTime? = null
) {
    @PrePersist
    fun onCreate() {
        creationTime = ZonedDateTime.now()
        updated = ZonedDateTime.now()
    }

    @PreUpdate
    fun onUpdate() {
        updated = ZonedDateTime.now()
    }
}